### PR TITLE
Link/anchor component

### PR DESCRIPTION
**Pull request by leneherng**
Add Link component
- Renamed as `Link` since its more familiar to people than `Anchor`
- Configured [SWC compliler](https://storybook.js.org/docs/configure/integration/compilers#the-swc-compiler-doesnt-work-with-react) to work with React
- Add react vite plugin [explicitly](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#framework-specific-vite-plugins-have-to-be-explicitly-added)

### CODE REVIEW

File: packages/react/src/components/button.tsx [hunk 1]
```diff
@@ -31,60 +32,58 @@ const button_cva = cva(
...
-          "focus:ring ring-fr-primary/0 focus:ring-fr-primary",
+          "focus:ring focus:ring-fr-danger",
           "disabled:bg-bg-white-disabled disabled:text-txt-danger-disabled disabled:border-transparent",
         ],
         "danger-ghost": [
           "bg-transparent border border-transparent text-txt-danger",
           "hover:bg-bg-danger-50",
-          "focus:ring ring-fr-primary/0 focus:ring-fr-primary",
+          "focus:ring focus:ring-fr-danger",
           "disabled:bg-bg-white-disabled disabled:text-txt-danger-disabled disabled:border-transparent",
```
Comment (irfancoder) [inline packages/react/src/components/button.tsx:60]: Why change to `ring-fr-danger`? The `fr-primary` is following the Figma spec: https://www.figma.com/design/svmWSPZarzWrJ116CQ8zpV/MYDS-(Beta)?node-id=11-11293&node-type=canvas&t=oijp5bsFSusOoHwK-0

File: packages/react/src/components/link.tsx
Comment (irfancoder) [inline packages/react/src/components/link.tsx:1]: Inherit `variant` & `size` props as part of `Link`. There will be cases where we make an `<a>` and dress it up as button.

Comment (irfancoder) [reply packages/react/src/components/link.tsx:1]: Also, just like you did for `button.tsx`, make an `asChild` option here. To cater to edge-case components: 

eg, Nextjs `Link` component

**Review by irfancoder — CHANGES_REQUESTED** [review]

File: packages/react/src/components/button.tsx (hunk 1 above)
Comment (leneherng) [reply packages/react/src/components/button.tsx:60]: ah my bad, was testing cause i thought it was kinda odd, will change it back to primary

File: packages/react/src/components/link.tsx
Comment (leneherng) [reply packages/react/src/components/link.tsx:1]: > Inherit `variant` & `size` props as part of `Link`. There will be cases where we make an `<a>` and dress it up as button.

Yea for those cases, was thinking of following [this](https://ui.shadcn.com/docs/components/button#link), utilising  Button asChild to nest the `<a>` like the Backlink story, wdyt?

Comment (leneherng) [reply packages/react/src/components/link.tsx:1]: > Also, just like you did for `button.tsx`, make an `asChild` option here. To cater to edge-case components:
> 
> eg, Nextjs `Link` component

ooh good idea! Will implement

**Review by irfancoder — APPROVED** [review]

**Pull request merged**